backend/fms_core/services/sample.py

```diff
@@ -525,6 +525,36 @@ def remove_sample_metadata(sample, metadata):
     return deleted, errors, warnings
 
 
+def validate_normalization(initial_volume, initial_concentration, final_volume, desired_concentration, tolerance=0.01):
+    is_valid = None
+    errors = []
+    warnings = []
+
+    # Validate parameters
+    if initial_volume is None:
+        errors.append(f"Initial volume is required to validate concentration.")
+    if initial_concentration is None:
+        errors.append(f"Initial concentration is invalid.")
+    if final_volume is None:
+        errors.append(f"Final volume is required for validation.")
+    if desired_concentration is None:
+        errors.append(f"Final concentration is required for validation.")
+
+    if not errors:
+        # Calculate the current amount to be able to calculate final concentration
+        solute_amount = initial_concentration * initial_volume
+        computed_concentration = solute_amount / final_volume
+
+        delta_concentration = computed_concentration - desired_concentration
+        if abs(delta_concentration) <= tolerance:
+            is_valid = True
+        else:
+            errors.append(f'Desired concentration [{desired_concentration}] is not valid.')
```

Comment:
i would add " given the dilution ratio (source volume used / final volume)." to make it clear that validation is linked to the volumes submitted in the template too.